### PR TITLE
feat(description): add files description in yaml

### PR DESCRIPTION
**Pull request by fzipi**
Signed-off-by: Felipe Zipitria <felipe.zipitria@owasp.org>

Generate file description based on yaml content, not in the markdown file. Played with the idea of adding a general comment to the rules file itself, so it is even easier to maintain. But then I've failed to easily parse that from hugo. So here is my second best approach.

Now we need to add the relevant information on the yaml file only!

### CODE REVIEW

File: data/filesdescription.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  RESPONSE-980-CORRELATION.conf: >
+    The rules in this configuration file facilitate the gathering of data
+    about successful and unsuccessful attacks on the server.
+  RESPONSE-999-EXCLUSION-RULES-AFTER-CRS.conf.example: >
+    This file is used to add LOCAL exceptions for your site. Often in this
+    file we would see rules that short-circuit inspection and allow
+    certain transactions to skip through inspection.
+  RESPONSE-980-CORRELATION.conf: TODO
+
+"3":
```
Comment (airween) [inline data/filesdescription.yaml:75]: Is this the version number? If yes, will it enough to represent the whole versioning of rule set?

Comment (fzipi) [reply data/filesdescription.yaml:75]: It is just a key. We can use more granular if things change in between minor versions.

**Comment by fzipi** [comment]
@RedXanadu @dune73 What do you think about this approach?

**Comment by theseion** [comment]
Neat!

**Comment by RedXanadu** [comment]
Sorry @fzipi, I'll admit, I had a quick look when you submitted this the other week and I didn't understand it. I haven't had time to properly look into it yet. Apologies.

**Comment by dune73** [comment]
I see more or less what you are doing here @fzipi. But I do not yet understand the motivation. Could you explain why we need this to be machine readable?

**Comment by fzipi** [comment]
To me, it's easier to maintain in a centralized place. Having multiple major versions might imply a headache on copy and pasting different files. Keeping the format also is a pain. This way you just write in one file and forget about the format.

Ideally, we could write the comment in the `.conf` file and parse it automagically to generate this content. But there is no easy way to do that.

**Comment by fzipi** [comment]
Any other comments here?

File: data/filesdescription.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
+---
+common:
+  REQUEST-900-EXCLUSION-RULES-BEFORE-CRS.conf.example: >
+    This file is used to add LOCAL exceptions for your site. Often in this file
+    we would see rules that short-circuit inspection and allow certain
+    transactions to skip through inspection.
```
Comment (theseion) [inline data/filesdescription.yaml:6]: In folded scalar style, an empty line is necessary to insert a break. I think the example should be on its own line.
```suggestion
    transactions to skip through inspection.
    
```

File: data/filesdescription.yaml (hunk 1 above)
Comment (theseion) [inline data/filesdescription.yaml:75]: I would like this to be more specific, e.g.,:
```suggestion
versioned:
  "3":
```

File: data/filesdescription.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+"3":
+  REQUEST-903.9001-DRUPAL-EXCLUSION-RULES.conf: TODO
+  REQUEST-903.9002-WORDPRESS-EXCLUSION-RULES.conf: TODO
+  REQUEST-903.9003-NEXTCLOUD-EXCLUSION-RULES.conf: TODO
+  REQUEST-903.9004-DOKUWIKI-EXCLUSION-RULES.conf: TODO
+  REQUEST-903.9005-CPANEL-EXCLUSION-RULES.conf: TODO
+  REQUEST-903.9006-XENFORO-EXCLUSION-RULES.conf: TODO
+  REQUEST-910-IP-REPUTATION.conf: >
+    These rules deal with detecting traffic from IPs that have previously been
+    involved with malicious activity, either on our local site or globally. |
```
Comment (theseion) [inline data/filesdescription.yaml:84]:
```suggestion
    involved with malicious activity, either on our local site or globally.
```

**Review by theseion — CHANGES_REQUESTED** [review]

File: layouts/shortcodes/describe-rules.html [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{{- $crsversion := .Get "version" -}}
+{{- $files := merge $.Site.Data.filesdescription.common (index $.Site.Data.filesdescription $crsversion)}}
```
Comment (fzipi) [inline layouts/shortcodes/describe-rules.html:2]:
```suggestion
{{- $files := merge $.Site.Data.filesdescription.common (index $.Site.Data.filesdescription.versioned $crsversion)}}
```

File: data/filesdescription.yaml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  REQUEST-903.9006-XENFORO-EXCLUSION-RULES.conf: TODO
+  REQUEST-910-IP-REPUTATION.conf: >
+    These rules deal with detecting traffic from IPs that have previously been
+    involved with malicious activity, either on our local site or globally.
+  REQUEST-912-DOS-PROTECTION.conf: >
+    The rules in this file will attempt to detect some level 7 DoS (Denial
+    of Service) attacks against your server.
+"4":
+  REQUEST-905-COMMON-EXCEPTIONS.conf: TODO
+  RESPONSE-955-WEB-SHELLS.conf: TODO
```
Comment (fzipi) [inline data/filesdescription.yaml:92]:
```suggestion
  "4":
    REQUEST-905-COMMON-EXCEPTIONS.conf: TODO
    RESPONSE-955-WEB-SHELLS.conf: TODO
```

File: data/filesdescription.yaml [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  REQUEST-903.9003-NEXTCLOUD-EXCLUSION-RULES.conf: TODO
+  REQUEST-903.9004-DOKUWIKI-EXCLUSION-RULES.conf: TODO
+  REQUEST-903.9005-CPANEL-EXCLUSION-RULES.conf: TODO
+  REQUEST-903.9006-XENFORO-EXCLUSION-RULES.conf: TODO
+  REQUEST-910-IP-REPUTATION.conf: >
+    These rules deal with detecting traffic from IPs that have previously been
+    involved with malicious activity, either on our local site or globally.
+  REQUEST-912-DOS-PROTECTION.conf: >
+    The rules in this file will attempt to detect some level 7 DoS (Denial
+    of Service) attacks against your server.
```
Comment (fzipi) [inline data/filesdescription.yaml:89]:
```suggestion
  "3":
    REQUEST-903.9001-DRUPAL-EXCLUSION-RULES.conf: TODO
    REQUEST-903.9002-WORDPRESS-EXCLUSION-RULES.conf: TODO
    REQUEST-903.9003-NEXTCLOUD-EXCLUSION-RULES.conf: TODO
    REQUEST-903.9004-DOKUWIKI-EXCLUSION-RULES.conf: TODO
    REQUEST-903.9005-CPANEL-EXCLUSION-RULES.conf: TODO
    REQUEST-903.9006-XENFORO-EXCLUSION-RULES.conf: TODO
    REQUEST-910-IP-REPUTATION.conf: >
      These rules deal with detecting traffic from IPs that have   previously been involved with malicious activity, either on our local site or globally.
    REQUEST-912-DOS-PROTECTION.conf: >
      The rules in this file will attempt to detect some level 7 DoS (Denial of Service) attacks against your server.
```

File: layouts/shortcodes/describe-rules.html (hunk 4 above)
Comment (theseion) [reply layouts/shortcodes/describe-rules.html:2]: Yes, indeed.

**Review by theseion — APPROVED** [review]

**Comment by theseion** [comment]
LGTM.

**Comment by dune73** [comment]
Agreed

**Pull request merged**